src/main.c

```diff
@@ -257,11 +253,19 @@ int main(int argc, char **argv)
 	srand(GetTickCount());
 #elif defined(__WATCOMC__)
 	srand(time(NULL));
+#elif defined(__amigaos4__)
+	struct TimeVal tv;
+	amiga_inittimer();
+	amiga_getsystime(&tv);
+	srand(tv.Microseconds);
+#elif defined(XMP_AMIGA)
+	struct timeval tv;
+	amiga_inittimer();
+	amiga_getsystime(&tv);
+	srand(tv.tv_micro);
 #else
 	struct timeval tv;
-	struct timezone tz;
-
-	gettimeofday(&tv, &tz);
+	gettimeofday(&tv, NULL);
```

Comment:
Would it be worth abstracting some this into a separate function in `delay.c`?

Reply:
Probably yes. (If you have a patch, send it in.)

Reply:
Only the Amiga portions of this seem to relate to `delay.c`, correct?

Reply:
I _suppose_ that's what Cameron meant, but doing that actually seem like a hassle.